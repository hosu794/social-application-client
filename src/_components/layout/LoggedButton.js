import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { authActions } from "../../_actions";
import PropTypes from "prop-types";
import loading from "../../img/loading.gif";
import { Link } from "react-router-dom";

import LogoutButton from "../auth/LogoutButton";

function LoggedButton() {
  const dispatch = useDispatch();

  const user = useSelector((state) => state.user.user);

  function logoutUserOnClick() {
    dispatch(authActions.logout());
  }

  return (
    <React.Fragment>
      {user ? (
        <div>
          <Link to="/account" className="button is-primary">
            <div className="level">
              <div
                className="level-item level-left "
                style={{ marginRight: "0.5rem" }}
              >
                <img
                  style={{
                    width: "2rem",
                    height: "2rem",
                    borderRadius: "10rem",
                  }}
                  src={
                    user.avatarDownloadUri
                      ? user.avatarDownloadUri
                      : "https://bulma.io/images/placeholders/128x128.png"
                  }
                />
                <div
                  style={{
                    position: "absolute",
                  }}
                ></div>
              </div>
            </div>
          </Link>
          <LogoutButton />
        </div>
      ) : (
        <div>
          <button className="button is-loading"></button>
          <LogoutButton />
        </div>
      )}
    </React.Fragment>
  );
}

LoggedButton.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.bool.isRequired,
    username: PropTypes.bool.isRequired,
    name: PropTypes.bool.isRequired,
  }),
  authActions: PropTypes.object,
};

export default LoggedButton;
