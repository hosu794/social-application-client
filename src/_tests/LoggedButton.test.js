import { shallow, mount, configure, render } from "enzyme";

import React from "react";

import sinon, { expectation } from "sinon";

import LoggedButton from "../_components/layout/LoggedButton";

import configureStore from "redux-mock-store";

import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Switch } from "antd";

const mockStore = configureStore([]);

describe("My connected React-Redux Component", () => {
  let store;
  let wrapper;

  beforeAll(() => {
    store = mockStore({
      user: {
        user: {
          id: 1,
          username: "username",
          name: "Joe Doe",
          avatarDownloadUri: "https://bulma.io/images/placeholders/128x128.png",
        },
      },
    });

    store.dispatch = jest.fn();

    wrapper = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <LoggedButton />
        </BrowserRouter>
      </Provider>
    );
  });

  test("should render component correctly", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test("should dispatch an action on button click", () => {});
});
