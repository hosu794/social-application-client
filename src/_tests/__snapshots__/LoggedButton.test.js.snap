// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My connected React-Redux Component should render component correctly 1`] = `
<div>
  <a
    className="button is-primary"
    href="/account"
    onClick={[Function]}
  >
    <div
      className="level"
    >
      <div
        className="level-item level-left "
        style={
          Object {
            "marginRight": "0.5rem",
          }
        }
      >
        <img
          src="https://bulma.io/images/placeholders/128x128.png"
          style={
            Object {
              "borderRadius": "10rem",
              "height": "2rem",
              "width": "2rem",
            }
          }
        />
        <div
          style={
            Object {
              "position": "absolute",
            }
          }
        />
      </div>
    </div>
  </a>
  <a
    className="button is-light"
    onClick={[Function]}
  >
    Log out
  </a>
</div>
`;
